crates/core/src/transport/connection_handler.rs

```diff
@@ -197,7 +198,7 @@ impl<S: Socket> UdpPacketsListener<S> {
 
     async fn gateway_connection(
         &mut self,
-        remote_intro_packet: PacketData,
+        remote_intro_packet: PacketData<AssymetricEncrypted>,
```

Comment:
we can add type aliases @ packet_data mod for making this easier to deal with, eg:
`pub(super) type AssymetricPacket = PacketData<AssymetricEncrypted>`